### PR TITLE
feat(ui-element): Add TextWithImageUiElement

### PR DESCRIPTION
**Pull request by welcoMattic**
This PR adds a hard dependency to [SyliusMediaManagerPlugin](https://github.com/monsieurbiz/SyliusMediaManagerPlugin/), to be able to use the best `ImageType`.

# Admin form
Screenshot 2024-08-06 at 11-12-37 Edit product Sylius

# Admin preview
Screenshot 2024-08-06 at 11-21-56 Edit product Sylius

### CODE REVIEW

File: src/Form/Type/UiElement/TextWithImageUiElementType.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                'label' => 'monsieurbiz_richeditor_plugin.ui_element.monsieurbiz.image.field.image',
+                'required' => true,
+                'with_link' => false,
+                'with_alignment' => false,
+            ])
+            ->add('imageAlign', ChoiceType::class, [
+                'label' => 'monsieurbiz_ui_elements.common.fields.alignment',
+                'required' => true,
+                'choices' => [
+                    \sprintf('monsieurbiz_ui_elements.ui_element.text_with_image_ui_element.fields.image_align.choices.%s', self::IMAGE_POSITION_LEFT) => self::IMAGE_POSITION_LEFT,
```
Comment (lanfisis) [inline src/Form/Type/UiElement/TextWithImageUiElementType.php:63]: **🎯 suggestion:** ‏
```suggestion
                    \sprintf('%s.%s', self::IMAGE_POSITION_LABEL, self::IMAGE_POSITION_LEFT) => self::IMAGE_POSITION_LEFT,
```
Add const for better readability

**Review by lanfisis — CHANGES_REQUESTED** [review]

File: src/Form/Type/ImageType.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        parent::configureOptions($resolver);
+        $resolver->setDefaults([
+            'with_link' => true,
+            'with_alignment' => true,
+        ]);
+        $resolver->setAllowedTypes('with_link', ['null', 'bool']);
+        $resolver->setAllowedTypes('with_alignment', ['null', 'bool']);
+    }
+}
```
Comment (delyriand) [inline src/Form/Type/ImageType.php:46]: **🎯 suggestion:** use the `getParent` method instead extends the RichEditorImageType class

‏
```suggestion
class ImageType extends AbstractType
{
    /**
     * @inheritdoc
     */
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        if (!$options['with_link']) {
            $builder->remove('link');
        }

        if (!$options['with_alignment']) {
            $builder->remove('align');
        }
    }

    public function configureOptions(OptionsResolver $resolver): void
    {
        $resolver->setDefaults([
            'with_link' => true,
            'with_alignment' => true,
        ]);
        $resolver->setAllowedTypes('with_link', ['null', 'bool']);
        $resolver->setAllowedTypes('with_alignment', ['null', 'bool']);
    }
    
    public function getParent(): string
    {
        return RichEditorImageType::class;
    }
}
```

**Review by delyriand — APPROVED** [review]

File: src/Form/Type/ImageType.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusUiElementsPlugin\Form\Type;
+
+use MonsieurBiz\SyliusRichEditorPlugin\Form\Type\UiElement\ImageType as RichEditorImageType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class ImageType extends AbstractType
```
Comment (maximehuran) [inline src/Form/Type/ImageType.php:19]: What is the use case of this class ?

Comment (welcoMattic) [reply src/Form/Type/ImageType.php:19]: In `TextWithImageUiElement`, we need an image field without link and alignment. However, the `ImageType` of RichEditor provide mandatory `link` and `alignment` fields. This child `ImageType` allow us to choose whether to have `link` and/or `alignment` fields.

**Review by maximehuran — APPROVED** [review]

**Review by lanfisis — APPROVED** [review]

**Pull request merged**